src/librustc_trans/trans/common.rs

```diff
@@ -213,8 +213,27 @@ pub fn type_needs_unwind_cleanup<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>, ty: Ty<
     }
 }
 
+
+/// Issue #22815: call the `FunctionContext::type_needs_drop` method
+/// instead, if possible; it is strictly more precise than this
+/// function.
 pub fn type_needs_drop<'tcx>(cx: &ty::ctxt<'tcx>, ty: Ty<'tcx>) -> bool {
```

Comment:
I'd prefer to remove this fn, or make it scoped to `FunctionContext::type_needs_drop` -- is there a reason we can't do this?